plugins/node/opentelemetry-instrumentation-pg/src/types.ts

```diff
@@ -16,6 +16,30 @@
 
 import * as pgTypes from 'pg';
 import * as pgPoolTypes from 'pg-pool';
+import type * as api from '@opentelemetry/api';
+import { InstrumentationConfig } from '@opentelemetry/instrumentation';
+
+export interface PgInstrumentationExecutionResponseHook {
+  (span: api.Span, data: pgTypes.QueryResult | pgTypes.QueryArrayResult): void;
+}
+
+export interface PgInstrumentationConfig extends InstrumentationConfig {
+  /**
+   * If true, additional information about query parameters and
+   * results will be attached (as `attributes`) to spans representing
+   * database operations.
+   */
+  enhancedDatabaseReporting?: boolean;
```

Comment:
I'd just rely on responseHook to be set and avoid permutations of the configuration that have no function like "isEnhanced && !responseHook". Toggling the enhancement get just function by removing the hook.

If you can justify the additional boolean here then at least please name it something more descriptive - I'd love to know exactly what the configuration option does from the name.

Reply:
ok I'll adjust it. Initially I wanted to keep aligned to the `enhancedDatabaseReporting` definition which is that the result is also collected (wasn't implemented yet)

Reply:
Indeed `enhancedDatabaseReporting` is a seperate option that some instrumentation share to add more data on the span's attribute. It was already there @Rauno56 (https://github.com/open-telemetry/opentelemetry-js-contrib/blob/main/plugins/node/opentelemetry-instrumentation-pg/src/utils.ts#L85)